tests/generate_crlf_checkin.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ "$1" == "" -o "$2" == "" ]; then
+	echo "usage: $0 inputfiles-directory directory [tempdir]"
+	exit 1
+fi
+
+input=$1
+output=$2
+tempdir=$3
+
+if [ ${input:1} != "/" ]; then
+	input="$PWD/$input"
+fi
+
+if [ ${output:1} != "/" ]; then
+	output="$PWD/$output"
+fi
+
+if [ "${tempdir}" != "" -a "${tempdir:1}" != "/" ]; then
+	tempdir="$PWD/$tempdir"
+fi
+
+set -u
+
+create_test_data() {
+	local input=$1
+	local output=$2
+	local tempdir=$3
+	local safecrlf=$4
+	local autocrlf=$5
+	local attr=$6
+
+	local destdir="${output}/safecrlf_${safecrlf},autocrlf_${autocrlf}"
+
+	if [ "$attr" != "" ]; then
+		local attrdir=`echo $attr | sed -e "s/ /,/g" | sed -e "s/=/_/g"`
+		destdir="${destdir},${attrdir}"
+	fi
+
+	if [ "$tempdir" = "" ]; then
+		tempdir="${output}/generate_crlf_${RANDOM}"
+	else
+		tempdir="${tempdir}/generate_crlf_${RANDOM}"
+	fi
+
+	echo "Generating ${destdir}"
+	mkdir -p "${destdir}"
+	mkdir -p "${tempdir}"
+
+	git init "${tempdir}"
+	if [ "$attr" != "" ]; then
+		echo "* ${attr}" > "${tempdir}/.gitattributes"
+	fi
+	cp "$input"/* "${tempdir}"
+	pushd "${tempdir}"
+		git config core.autocrlf ${autocrlf}
+		git config core.safecrlf ${safecrlf}
+		for file in *
+		do
+			process_file "$destdir" "$file"
+		done
+	popd
+
+	rm -rf "$tempdir"
+}
+
+function process_file() {
+	destdir=$1
+	file=$2
+
+	rm -f "$destdir/$file.obj" "$destdir/$file.fail"
+
+	set +e
+	OUTPUT=$(git add "$file" 2>&1)
+	if [ $? -ne 0 ]; then
+		set -e
+		touch "$destdir/$file.fail"
+		if [ "${OUTPUT:0:38}" == "fatal: CRLF would be replaced by LF in" ]; then
+			echo "CRLF would be replaced by LF" > "$destdir/$file.fail"
+		elif [ "${OUTPUT:0:38}" == "fatal: LF would be replaced by CRLF in" ]; then
+			echo "LF would be replaced by CRLF" > "$destdir/$file.fail"
+		fi
+	else
+		OBJ=$(ls .git/objects/??/* 2>/dev/null)
+ 		set -e
+		OBJ=${OBJ:13} # 13=".git/objects/"
+		OBJ=${OBJ//\//}
+		echo "$OBJ" > "$destdir/$file.obj"
+	fi
+	rm -f .git/objects/??/*
+	rm -f .git/index
```

Comment:
In my first attempt all the private data was hardcoded, then it was requested that a script should be used for generating the "known good test results". I'm accessing the private data so that I can be sure which objects (ids) were created on adding a file (after the CRLF handling).

If you have a better idea here, I'm open.